### ISSUE TITLE
Add a lint check for usage of `when` in Kotlin tests

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#10449]
Good first issue: add a lint check for `when` to improve our Kotlin migration automation.

_Originally posted by @david-allison in https://github.com/ankidroid/Anki-Android/pull/10437#issuecomment-1060015668_

----

In Kotlin, `when` is a keyword, so method calls need a backtick around the method name. `whenever` is a fixed version of this method.

See: https://github.com/ankidroid/Anki-Android/blob/main/lint-rules/src/main/java/com/ichi2/anki/lint/rules/JUnitNullAssertionDetector.kt for how to write a lint check to ensure a method isn't called in Kotlin

**Comment by yashsharma81** [comment]
Hi @david-allison i want to work on this issue.

**Comment by david-allison** [comment]
Sounds good to me!

**Comment by david-allison** [comment]
Note: this can likely be implemented by adding:

`<issue id="NoHardKeywords" severity="fatal"/>` to `lint-release.xml`

**Comment by yashsharma81** [comment]
@david-allison could you please acknowledge me for fixing this issue, i'm new to Open Source Contributing..
Hoping for reply.

**Comment by david-allison** [comment]
Sorry, I don't quite understand what you're asking there. Could you rephrase the question?

**Comment by yashsharma81** [comment]
@david-allison how to "Add a lint check for usage of when in Kotlin test" i'm unable to do as i'm new to open source.

**Comment by david-allison** [comment]
> Note: this can likely be implemented by adding:
> 
> `<issue id="NoHardKeywords" severity="fatal"/>` to `lint-release.xml`

The full link to the file is: https://github.com/ankidroid/Anki-Android/blob/main/lint-release.xml

**Comment by yashsharma81** [comment]
@david-allison done please check https://github.com/ankidroid/Anki-Android/pull/10660

**Comment by yashsharma81** [comment]
still working but can't find the solution
